File: languages/cpp/01-foundations/arrays-and-vectors/example/vector-filter.cpp
#include <iostream>
#include <vector>

using namespace std;

int main() {
    int n = 0;
    cout << "How many numbers? ";
    cin >> n;

    if (n <= 0) {
        cout << "Please enter a positive count.\n";
        return 0;
    }

    vector<int> values;
    values.reserve(static_cast<size_t>(n));

    for (int i = 0; i < n; ++i) {
        int value = 0;
        cout << "Value " << (i + 1) << ": ";
        cin >> value;
        values.push_back(value);
    }

    int threshold = 0;
    cout << "Filter values greater than or equal to: ";
    cin >> threshold;

    cout << "Filtered values: ";
    bool printedAny = false;
    for (size_t i = 0; i < values.size(); ++i) {
        if (values[i] >= threshold) {
            if (printedAny) {
                cout << ", ";
            }
            cout << values[i];
            printedAny = true;
        }
    }

    if (!printedAny) {
        cout << "none";
    }
    cout << '\n';

    return 0;
}

File: languages/python/01-foundations/arrays-and-vectors/example/vector-filter.py
# This helper example focuses on isolating one collection transformation so the filtering rule
# stands out.

# This extra example extends arrays-and-vectors with a focused filtering task.

raw_count = input("How many numbers? ").strip()
try:
    count = int(raw_count)
except ValueError:
    print("Please enter a positive count.")
else:
    if count <= 0:
        print("Please enter a positive count.")
    else:
        values = []
        for index in range(count):
            raw_value = input(f"Value {index + 1}: ").strip()
            try:
                values.append(int(raw_value))
            except ValueError:
                print("Please enter valid integers only.")
                break
        else:
            raw_threshold = input("Filter values greater than or equal to: ").strip()
            try:
                threshold = int(raw_threshold)
            except ValueError:
                print("Please enter a valid threshold.")
            else:
                filtered = [value for value in values if value >= threshold]
                if filtered:
                    print("Filtered values: " + ", ".join(str(value) for value in filtered))
                else:
                    print("Filtered values: none")
